pkg/validate/dynamic/dynamic.go

```diff
@@ -780,13 +772,31 @@ func (dv *dynamic) ValidatePreConfiguredNSGs(ctx context.Context, oc *api.OpenSh
 	return nil
 }
 
+func (dv *dynamic) validateActions(ctx context.Context, r *azure.Resource, actions []string) (*string, error) {
```

Comment:
I recommend adding a doc comment on this function, especially to make it easier for readers to understand what the string pointer return value is for.